src/components/topNav/TopNav.tsx

```diff
@@ -1,46 +1,45 @@
 import * as React from "react";
-import * as styles from "./TopNav.module.css";
 import { Link } from "@gemeente-denhaag/components-react";
-import { navigate } from "gatsby";
-
-export interface ITopNavItem {
-  label: string;
-  href: string;
-  icon?: JSX.Element;
-}
+import * as styles from "./TopNav.module.css";
+import clsx from "clsx";
 
-interface TopNavItemsProps {
-  items: ITopNavItem[];
+interface TopNavProps {
+  items: { label: string; icon?: JSX.Element; handleClick: () => any }[];
+  layoutClassName?: string;
 }
 
-export const PrimaryTopNav: React.FC<TopNavItemsProps> = ({ items }) => {
+export const PrimaryTopNav: React.FC<TopNavProps> = ({ items, layoutClassName }) => {
   return (
-    <nav className={styles.primary}>
-      <ul className={styles.ul}>
-        {items.map(({ label, href, icon }, idx) => (
-          <li className={styles.li} key={idx} onClick={() => navigate(href)}>
-            <Link icon={icon} iconAlign="start">
-              {label}
-            </Link>
-          </li>
-        ))}
-      </ul>
-    </nav>
+    <div className={clsx(styles.primary, layoutClassName && layoutClassName)}>
+      <nav className={styles.primary}>
+        <ul className={styles.ul}>
+          {items.map(({ label, icon, handleClick }, idx) => (
+            <li className={styles.li} key={idx} onClick={() => handleClick()}>
+              <Link className={styles.link} icon={icon} iconAlign="start">
+                {label}
+              </Link>
+            </li>
+          ))}
+        </ul>
+      </nav>
+    </div>
   );
 };
 
-export const SecondaryTopNav: React.FC<TopNavItemsProps> = ({ items }) => {
+export const SecondaryTopNav: React.FC<TopNavProps> = ({ items, layoutClassName }) => {
   return (
-    <nav>
-      <ul className={styles.ul}>
-        {items.map(({ label, href, icon }, idx) => (
-          <li className={styles.li} key={idx} onClick={() => navigate(href)}>
-            <Link icon={icon} iconAlign="start">
-              {label}
-            </Link>
-          </li>
-        ))}
-      </ul>
-    </nav>
+    <div className={clsx(styles.secondary, layoutClassName && layoutClassName)}>
+      <nav>
+        <ul className={styles.ul}>
+          {items.map(({ label, icon, handleClick }, idx) => (
+            <li className={styles.li} key={idx} onClick={() => handleClick}>
```

Comment:
Passing a function to this onClick prop can be done without explicitly calling it, like so:

```
onClick={handleClick}
```